lib/s3.js

```diff
@@ -31,6 +31,13 @@ module.exports = CoreObject.extend({
     var config = plugin.pluginConfig;
 
     this._plugin = plugin;
+
+    const profile = this._plugin.readConfig('profile');
```

Comment:
can we please move this up to the declaration of the other vars and declare as var please for consistency? I suggest we move to es6 syntax in a follow up pr.